src/pudl/analysis/allocate_net_gen.py

```diff
@@ -645,30 +1135,501 @@ def calc_allocation_fraction(gen_pm_fuel, drop_interim_cols=True):
+        # report in the boiler fuel table
+        frac_from_bf_tbl=lambda x: x.fuel_consumed_mmbtu_bf_tbl_pm_fuel
+        / x.fuel_consumed_mmbtu_gf_tbl,
+        # for records within these mix groups that do have fuel consumption in the
+        # bf table..
+        frac_fuel=lambda x: x.fuel_consumed_mmbtu_bf_tbl
+        / x.fuel_consumed_mmbtu_bf_tbl_pm_fuel,  # generator based fuel from bf table
+        frac_bf=lambda x: x.frac_fuel * x.frac_from_bf_tbl,
+        # fraction of fuel that does not show up in the bf table
+        # set minimum fraction to zero so we don't get negative fuel
+        frac_missing_from_bf_tbl=lambda x: np.where(
+            (x.frac_from_bf_tbl < 1), (1 - x.frac_from_bf_tbl), 0
+        ),
+        capacity_mw_missing_from_bf_tbl=lambda x: np.where(
+            x.in_bf_tbl, 0, x.capacity_mw
+        ),
+        frac_cap=lambda x: x.frac_missing_from_bf_tbl
+        * (x.capacity_mw_missing_from_bf_tbl / x.capacity_mw_fuel_in_bf_tbl_group),
+        # the real deal
+        # this could aslo be `x.frac_bf + x.frac_cap` because the frac_bf
+        # should be 0 for any generator that does not have fuel in the bf_tbl
+        # and frac_cap should be 0 for any generator that has fuel in the
+        # bf_tbl.
+        frac=lambda x: np.where(x.in_bf_tbl, x.frac_bf, x.frac_cap),
+    )
+    # _ = _test_frac(some_bf)
+
+    # Calculate what fraction of the total capacity is associated with each of
+    # the generators in the grouping.
+    gf_only = gf_only.assign(
+        frac_cap=lambda x: x.capacity_mw / x.capacity_mw_unit_fuel,
+        frac=lambda x: x.frac_cap,
+    )
+    # _ = _test_frac(gf_only)
+
+    # squish all of these methods back together.
+    fuel_alloc = pd.concat([all_bf, some_bf, gf_only])
+    # _ = _test_frac(fuel_alloc)
+
+    # replace the placeholder missing values with zero before allocating
+    # since some of these may have been aggregated, well, flag any values less than the
+    # sentinel missing value
+    fuel_alloc = fuel_alloc.assign(
+        fuel_consumed_mmbtu_gf_tbl=remove_aggregated_sentinel_value(
+            fuel_alloc["fuel_consumed_mmbtu_gf_tbl"]
+        ),
+        fuel_consumed_for_electricity_mmbtu_gf_tbl=remove_aggregated_sentinel_value(
+            fuel_alloc["fuel_consumed_for_electricity_mmbtu_gf_tbl"]
+        ),
+    )
+    # do the allocating-ing!
+    fuel_alloc = (
+        fuel_alloc.assign(
+            # we could x.fuel_consumed_mmbtu_bf_tbl.fillna here if we wanted to
+            # take the net gen
+            fuel_consumed_mmbtu=lambda x: x.fuel_consumed_mmbtu_gf_tbl * x.frac,
+            fuel_consumed_for_electricity_mmbtu=lambda x: x.fuel_consumed_for_electricity_mmbtu_gf_tbl
+            * x.frac,
+        )
+        .pipe(apply_pudl_dtypes, group="eia")
+        .dropna(how="all")
+    )
+
+    return fuel_alloc
+
+
+def remove_aggregated_sentinel_value(col: pd.Series, scalar: int = 20) -> pd.Series:
```

Comment:
I don't think `scalar` is really an integer here (i.e. a count of indivisible things).

Reply:
what do you think it is then?